Validate community description async for token owner communities

cammellos opened the pull request:
This commit changes the behavior of the system so that if a community description with a token owner is detected, it will queue it up and validates it in the background. 
It will check ownership against the contract and if the ownership validates, it will process the description and signal the user.

I am missing one test at the messenger level to validate the behavior, I will be adding that, but in the meantime it would be good to get some eyes on it.
There's also an issue with out of order messages (that's pre-existing), when a community description will arrive after some change of membership occurred. We talked about re-fetching from mailserver, but I would address this separately as it's connected as well on how communities are encrypted, I will pick it up next. 

This will have to be throughlty tested and validated, I might need some help setting things up.

Built on Patryk's code as you can see from the first commit.

osmaczko committed:
Add queue for processing community description & configurable control node

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (100)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 24ee5cb2 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/1/) | 2023-10-05 13:47:28 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/1//consoleText) |
| :heavy_check_mark: | 24ee5cb2 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/1/) | 2023-10-05 13:48:25 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-134449-24ee5c-pr4108.zip) |
| :heavy_check_mark: | 24ee5cb2 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/1/) | 2023-10-05 13:49:15 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-134449-24ee5c-pr4108.zip) |
| :heavy_check_mark: | 24ee5cb2 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/1/) | 2023-10-05 13:49:43 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-134449-24ee5c-pr4108.aar) |
| | | | | | | |
| :heavy_multiplication_x: | c81bdd00 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/2/) | 2023-10-05 14:30:20 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/2//consoleText) |
| :heavy_check_mark: | c81bdd00 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/2/) | 2023-10-05 14:30:23 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-142914-c81bdd-pr4108.zip) |
| :heavy_check_mark: | c81bdd00 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/2/) | 2023-10-05 14:31:57 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-142910-c81bdd-pr4108.zip) |
| :heavy_check_mark: | c81bdd00 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/2/) | 2023-10-05 14:34:10 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-142910-c81bdd-pr4108.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 197e2b4c | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/3/) | 2023-10-05 15:19:26 | ~59 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/3//consoleText) |
| :heavy_check_mark: | 197e2b4c | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/3/) | 2023-10-05 15:19:46 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-151824-197e2b-pr4108.zip) |
| :heavy_check_mark: | 197e2b4c | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/3/) | 2023-10-05 15:20:06 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-151822-197e2b-pr4108.aar) |
| :heavy_check_mark: | 197e2b4c | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/3/) | 2023-10-05 15:21:58 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-151823-197e2b-pr4108.zip) |
| | | | | | | |
| :heavy_check_mark: | 0f2d1491 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/4/) | 2023-10-05 15:29:17 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-152705-0f2d14-pr4108.aar) |
| :heavy_check_mark: | 0f2d1491 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/4/) | 2023-10-05 15:29:54 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-152705-0f2d14-pr4108.zip) |
| :heavy_check_mark: | 0f2d1491 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/4/) | 2023-10-05 15:35:00 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-152705-0f2d14-pr4108.zip) |
| :heavy_multiplication_x: | 0f2d1491 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/4/) | 2023-10-05 15:38:10 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ce18a291 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/5/) | 2023-10-05 16:00:18 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-155825-ce18a2-pr4108.aar) |
| :heavy_check_mark: | ce18a291 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/5/) | 2023-10-05 16:01:17 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-155825-ce18a2-pr4108.zip) |
| :heavy_check_mark: | ce18a291 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/5/) | 2023-10-05 16:01:41 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-155825-ce18a2-pr4108.zip) |
| :heavy_multiplication_x: | ce18a291 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/5/) | 2023-10-05 16:07:36 | ~9 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5ac7a5c5 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/6/) | 2023-10-05 16:23:35 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-162155-5ac7a5-pr4108.zip) |
| :heavy_check_mark: | 5ac7a5c5 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/6/) | 2023-10-05 16:23:45 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-162155-5ac7a5-pr4108.aar) |
| :heavy_check_mark: | 5ac7a5c5 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/6/) | 2023-10-05 16:24:39 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-162155-5ac7a5-pr4108.zip) |
| :heavy_multiplication_x: | 5ac7a5c5 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/6/) | 2023-10-05 16:30:04 | ~8 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | cd815b3f | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/7/) | 2023-10-06 08:45:38 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231006-084339-cd815b-pr4108.aar) |
| :heavy_check_mark: | cd815b3f | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/7/) | 2023-10-06 08:46:32 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231006-084339-cd815b-pr4108.zip) |
| :heavy_check_mark: | cd815b3f | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/7/) | 2023-10-06 08:46:50 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231006-084339-cd815b-pr4108.zip) |
| :heavy_multiplication_x: | cd815b3f | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/7/) | 2023-10-06 08:51:26 | ~7 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/7//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | daf29ef3 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/8/) | 2023-10-10 09:41:22 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/8//consoleText) |
| :heavy_check_mark: | daf29ef3 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/8/) | 2023-10-10 09:41:51 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231010-094016-daf29e-pr4108.zip) |
| :heavy_check_mark: | daf29ef3 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/8/) | 2023-10-10 09:42:22 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231010-094016-daf29e-pr4108.aar) |
| :heavy_check_mark: | daf29ef3 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/8/) | 2023-10-10 09:43:31 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231010-094016-daf29e-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 6e545d68 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/9/) | 2023-10-10 09:42:39 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/9//consoleText) |
| :heavy_check_mark: | 6e545d68 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/9/) | 2023-10-10 09:43:35 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231010-094154-6e545d-pr4108.zip) |
| :heavy_check_mark: | 6e545d68 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/9/) | 2023-10-10 09:44:36 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231010-094227-6e545d-pr4108.aar) |
| :heavy_check_mark: | 6e545d68 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/9/) | 2023-10-10 09:46:42 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231010-094334-6e545d-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 00c94b35 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/10/) | 2023-10-10 10:26:24 | ~58 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/10//consoleText) |
| :heavy_check_mark: | 00c94b35 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/10/) | 2023-10-10 10:26:56 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231010-102526-00c94b-pr4108.zip) |
| :heavy_check_mark: | 00c94b35 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/10/) | 2023-10-10 10:27:33 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231010-102522-00c94b-pr4108.aar) |
| :heavy_check_mark: | 00c94b35 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/10/) | 2023-10-10 10:28:17 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231010-102522-00c94b-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 9a5e8604 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/11/) | 2023-10-10 10:56:41 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/11//consoleText) |
| :heavy_check_mark: | 9a5e8604 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/11/) | 2023-10-10 10:57:01 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231010-105527-9a5e86-pr4108.zip) |
| :heavy_check_mark: | 9a5e8604 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/11/) | 2023-10-10 10:57:36 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231010-105527-9a5e86-pr4108.aar) |
| :heavy_check_mark: | 9a5e8604 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/11/) | 2023-10-10 10:58:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231010-105527-9a5e86-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | d062cb73 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/12/) | 2023-10-10 11:30:38 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/12//consoleText) |
| :heavy_check_mark: | d062cb73 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/12/) | 2023-10-10 11:31:06 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231010-112937-d062cb-pr4108.zip) |
| :heavy_check_mark: | d062cb73 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/12/) | 2023-10-10 11:31:20 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231010-112937-d062cb-pr4108.aar) |
| :heavy_check_mark: | d062cb73 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/12/) | 2023-10-10 11:32:29 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231010-112937-d062cb-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 4ddf6efa | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/13/) | 2023-10-13 06:37:17 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/13//consoleText) |
| :heavy_check_mark: | 4ddf6efa | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/13/) | 2023-10-13 06:37:26 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-063538-4ddf6e-pr4108.zip) |
| :heavy_check_mark: | 4ddf6efa | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/13/) | 2023-10-13 06:37:59 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-063534-4ddf6e-pr4108.aar) |
| :heavy_check_mark: | 4ddf6efa | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/13/) | 2023-10-13 06:40:24 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-063538-4ddf6e-pr4108.zip) |
| | | | | | | |
| :heavy_check_mark: | 1a44953f | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/14/) | 2023-10-13 13:40:13 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-133756-1a4495-pr4108.aar) |
| :heavy_multiplication_x: | 1a44953f | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/14/) | 2023-10-13 13:40:16 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/14//consoleText) |
| :heavy_check_mark: | 1a44953f | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/14/) | 2023-10-13 13:40:57 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-133800-1a4495-pr4108.zip) |
| :heavy_check_mark: | 1a44953f | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/14/) | 2023-10-13 13:41:15 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-133756-1a4495-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | e1fd0f22 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/15/) | 2023-10-13 13:46:27 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/15//consoleText) |
| :heavy_check_mark: | e1fd0f22 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/15/) | 2023-10-13 13:46:59 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-134521-e1fd0f-pr4108.zip) |
| :heavy_check_mark: | e1fd0f22 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/15/) | 2023-10-13 13:47:09 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-134521-e1fd0f-pr4108.aar) |
| :heavy_check_mark: | e1fd0f22 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/15/) | 2023-10-13 13:48:09 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-134521-e1fd0f-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 7a27b96c | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/16/) | 2023-10-13 14:07:33 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/16//consoleText) |
| :heavy_check_mark: | 7a27b96c | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/16/) | 2023-10-13 14:07:58 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-140625-7a27b9-pr4108.zip) |
| :heavy_check_mark: | 7a27b96c | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/16/) | 2023-10-13 14:08:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-140621-7a27b9-pr4108.aar) |
| :heavy_check_mark: | 7a27b96c | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/16/) | 2023-10-13 14:09:57 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-140625-7a27b9-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | c6783e60 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/17/) | 2023-10-13 14:10:59 | ~53 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/17//consoleText) |
| :heavy_check_mark: | c6783e60 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/17/) | 2023-10-13 14:11:19 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-141005-c6783e-pr4108.zip) |
| :heavy_check_mark: | c6783e60 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/17/) | 2023-10-13 14:12:11 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-141000-c6783e-pr4108.aar) |
| :heavy_check_mark: | c6783e60 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/17/) | 2023-10-13 14:13:07 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-141001-c6783e-pr4108.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 306c6f08 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/18/) | 2023-10-13 14:11:58 | ~54 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/18//consoleText) |
| :heavy_check_mark: | 306c6f08 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/18/) | 2023-10-13 14:12:45 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-141123-306c6f-pr4108.zip) |
| :heavy_check_mark: | 306c6f08 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/18/) | 2023-10-13 14:13:52 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-141214-306c6f-pr4108.aar) |
| :heavy_check_mark: | 306c6f08 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/18/) | 2023-10-13 14:16:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-141311-306c6f-pr4108.zip) |
| | | | | | | |
| :heavy_check_mark: | 0da0ad4e | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/19/) | 2023-10-13 14:23:53 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-142226-0da0ad-pr4108.zip) |
| :heavy_check_mark: | 0da0ad4e | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/19/) | 2023-10-13 14:24:16 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-142226-0da0ad-pr4108.aar) |
| :heavy_check_mark: | 0da0ad4e | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/19/) | 2023-10-13 14:25:14 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-142226-0da0ad-pr4108.zip) |
| :heavy_multiplication_x: | 0da0ad4e | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/19/) | 2023-10-13 14:28:13 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/19//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 42045cb3 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/20/) | 2023-10-13 14:48:02 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-144625-42045c-pr4108.zip) |
| :heavy_check_mark: | 42045cb3 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/20/) | 2023-10-13 14:48:30 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-144621-42045c-pr4108.aar) |
| :heavy_check_mark: | 42045cb3 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/20/) | 2023-10-13 14:51:09 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-144621-42045c-pr4108.zip) |
| :heavy_multiplication_x: | 42045cb3 | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/20/) | 2023-10-13 15:19:19 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/20//consoleText) |
| | | | | | | |
| :heavy_check_mark: | be17600d | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/21/) | 2023-10-13 15:00:18 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-145856-be1760-pr4108.zip) |
| :heavy_check_mark: | be17600d | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/21/) | 2023-10-13 15:00:31 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-145856-be1760-pr4108.aar) |
| :heavy_check_mark: | be17600d | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/21/) | 2023-10-13 15:03:35 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-145856-be1760-pr4108.zip) |
| :heavy_multiplication_x: | be17600d | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/21/) | 2023-10-13 15:32:02 | ~12 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/21//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 29002a72 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/22/) | 2023-10-16 06:24:26 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231016-062252-29002a-pr4108.zip) |
| :heavy_check_mark: | 29002a72 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/22/) | 2023-10-16 06:24:57 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231016-062252-29002a-pr4108.aar) |
| :heavy_check_mark: | 29002a72 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/22/) | 2023-10-16 06:26:19 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231016-062252-29002a-pr4108.zip) |
| :heavy_multiplication_x: | 29002a72 | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/22/) | 2023-10-16 06:36:44 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/22//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 01e135c6 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/23/) | 2023-10-16 07:41:36 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231016-073957-01e135-pr4108.zip) |
| :heavy_check_mark: | 01e135c6 | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/23/) | 2023-10-16 07:42:00 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231016-073957-01e135-pr4108.aar) |
| :heavy_check_mark: | 01e135c6 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/23/) | 2023-10-16 07:43:00 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231016-073957-01e135-pr4108.zip) |
| :heavy_multiplication_x: | 01e135c6 | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/23/) | 2023-10-16 07:54:18 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/23//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 90646aac | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/24/) | 2023-10-17 06:26:49 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-062358-90646a-pr4108.zip) |
| :heavy_check_mark: | 90646aac | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/24/) | 2023-10-17 06:27:31 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-062358-90646a-pr4108.zip) |
| :heavy_check_mark: | 90646aac | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/24/) | 2023-10-17 06:29:07 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-062358-90646a-pr4108.aar) |
| :heavy_multiplication_x: | 90646aac | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/24/) | 2023-10-17 06:40:11 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/24//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e093fbfe | [#25](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/25/) | 2023-10-17 07:14:31 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-071254-e093fb-pr4108.zip) |
| :heavy_check_mark: | e093fbfe | [#25](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/25/) | 2023-10-17 07:14:52 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-071254-e093fb-pr4108.aar) |
| :heavy_check_mark: | e093fbfe | [#25](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/25/) | 2023-10-17 07:16:10 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-071254-e093fb-pr4108.zip) |
| :heavy_multiplication_x: | e093fbfe | [#25](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/25/) | 2023-10-17 07:38:32 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/25//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 1b144bfa | [#26](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/26/) | 2023-10-17 15:43:19 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-154102-1b144b-pr4108.aar) |
| :heavy_check_mark: | 1b144bfa | [#26](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/26/) | 2023-10-17 15:44:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-154102-1b144b-pr4108.zip) |
| :heavy_check_mark: | 1b144bfa | [#26](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/26/) | 2023-10-17 15:44:13 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-154102-1b144b-pr4108.zip) |
| :heavy_multiplication_x: | 1b144bfa | [#26](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/26/) | 2023-10-17 15:52:13 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/26//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 82f86c71 | [#27](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4108/27/) | 2023-10-17 16:50:06 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-164832-82f86c-pr4108.zip) |
| :heavy_check_mark: | 82f86c71 | [#27](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4108/27/) | 2023-10-17 16:50:54 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-164832-82f86c-pr4108.aar) |
| :heavy_check_mark: | 82f86c71 | [#27](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4108/27/) | 2023-10-17 16:52:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-164832-82f86c-pr4108.zip) |
| :heavy_multiplication_x: | 82f86c71 | [#27](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/27/) | 2023-10-17 17:19:53 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/27//consoleText) |
| :heavy_check_mark: | 82f86c71 | [#28](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/28/) | 2023-10-17 18:36:12 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4108/28//consoleText) |

osmaczko commented on api/defaults.go:
```diff
@@ -26,6 +26,20 @@ const defaultKeycardPairingDataFile = "/ethereum/mainnet_rpc/keycard/pairings.js
+	wakuv2TestFleet = "wakuv2.test"
+)
+
+var defaultWakuNodes = map[string][]string{
```
Rebase gone wrong? :thinking: Seems unrelated to the changes

cammellos replied:
Sorry, yes, I have added some changes I had locally, will remove

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -348,19 +360,93 @@ func (m *Manager) Start() error {
 			select {
 			case <-m.quit:
-				m.logger.Debug("quitting ens verification loop")
+				m.logger.Debug("quitting owner verification loop")
```
```suggestion
				m.logger.Debug("quitting ens verification loop")
```

jrainville replied:
I think this log is still ens

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -348,19 +360,93 @@ func (m *Manager) Start() error {
+		for {
+			select {
+			case <-m.quit:
+				m.logger.Debug("quitting ens verification loop")
```
```suggestion
				m.logger.Debug("quitting owner verification loop")
```

jrainville replied:
this one should say owner

jrainville commented on protocol/communities/manager.go:
```diff
@@ -348,19 +360,93 @@ func (m *Manager) Start() error {
+						chainID := CommunityDescriptionTokenOwnerChainID(description)
+						if chainID == 0 {
+							// This should not happen
+							m.logger.Error("chain id is 0, ignoring")
```
Should you `continue` here?

cammellos replied:
Yes :)

jrainville commented on protocol/communities/manager.go:
```diff
@@ -348,19 +360,93 @@ func (m *Manager) Start() error {
+							continue
+						}
+						if response != nil {
+							m.publish(&Subscription{Community: response.Community, TokenCommunityValidated: response})
```
Just to be sure, this `publish` is local only right? It doesn't publish to peers?

cammellos replied:
Correct, it publishes only to `messenger` that can handle the message then and signal to the client

jrainville commented on protocol/communities/manager.go:
```diff
@@ -623,6 +709,23 @@ func (m *Manager) ControlledCommunities() ([]*Community, error) {
 }
 
+func (m *Manager) Owned() ([]*Community, error) {
+	communities, err := m.persistence.CommunitiesWithPrivateKey(&m.identity.PublicKey)
```
This function name makes it sound like we it only returns communities for which we have the community's private key. Is that the case?
If so, maybe there is a problem, because you can be the owner of a community because of the owner token while not having its private key.

Maybe this is all just a little naming problem

cammellos replied:
Yes, you would, but you would have the private key of the community, or rather, the contract will be set to your (or any public key you have access), and then you can `SetPrivateKey` of the community. 
So the function is checking that the community has private key set, and that you are the owner of that community (i.e the controlNode is the public key corresponding to the private key you have set on the community)
hope that it's clear, not sure if I explained it very clearly :)

jrainville replied:
Ok I get it. The name is  a bit confusing when you know the old way of doing things, but it's ok

jrainville commented on protocol/communities/manager.go:
I think @osmaczko 's PR was merged, so maybe this can be removed?

osmaczko replied:
Yes, this should be removed.

jrainville commented on protocol/communities/manager.go:
Do you plan to implement this in this PR or should we open a new issue and do it after?

cammellos replied:
oh, it's implemented in a different function, can be removed, good point

jrainville reviewed (commented):
Partial review because i need to take a break. 

This is massive work. Good job.

I'll finish reviewing later.

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -1325,21 +1462,48 @@ func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, des
+
+	// Override verified owner
+	if verifiedOwner != nil {
+		community.config.ControlNode = verifiedOwner
```
```suggestion
		community.setControlNode(verifiedOwner)
```

osmaczko commented on protocol/communities/manager_test.go:
```diff
@@ -1538,3 +1547,350 @@ func (s *ManagerSuite) buildCommunityWithChat() (*Community, string, error) {
+			s.FailNow("no subscription")
+		}
+	}
+
```
IMO `community.ControlNode() == owner`` would be worth checking here as well.

jrainville commented on protocol/communities/manager.go:
```diff
@@ -360,7 +374,80 @@ func (m *Manager) runENSVerificationLoop() {
+
+						m.logger.Info("validating community", zap.String("id", types.EncodeHex(communityToValidate.id)), zap.String("signer", common.PubkeyToHex(signer)))
 
+						ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
```
Is this a 3 seconds "sleep"? What is it for?

osmaczko replied:
It is a timeout for the contract call.

jrainville reviewed (commented):
Second round of reviews done. Turns out most of the remaining files were protobufs, so it was pretty easy to review :smile: 

Anyway, it all looks very good. Nice work

osmaczko reviewed (commented):
Very nice! Thanks for providing solid tests :ok_hand:

mprakhov commented on protocol/communities/community.go:
```diff
@@ -1255,6 +1256,22 @@ func (o *Community) PrivateKey() *ecdsa.PrivateKey {
+	return o.config.ControlNode
+}
+
+func (o *Community) setControlNode(pubKey *ecdsa.PublicKey) {
+	if pubKey != nil {
+		o.config.ControlNode = pubKey
+	}
+}
```
will this be changed? In my understanding, control node can be only one device, but in this case all devices will be treated as a control node because they all will have the same pubkey

cammellos replied:
I believe that's fine @osmaczko is working on some changes that would allow this. The key is set to the chat key of the user, so it's already shared across devices.

mprakhov reviewed (commented):
Great work!

mprakhov commented on protocol/communities/manager.go:
```diff
@@ -1310,12 +1417,42 @@ func (m *Manager) DeleteCategory(request *requests.DeleteCommunityCategory) (*Co
 }
 
-func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, description *protobuf.CommunityDescription, payload []byte) (*CommunityResponse, error) {
+func (m *Manager) Queue(signer *ecdsa.PublicKey, community *Community, description *protobuf.CommunityDescription, payload []byte) error {
```
description is not used here

mprakhov commented on protocol/communities/manager.go:
```diff
@@ -1310,12 +1417,42 @@ func (m *Manager) DeleteCategory(request *requests.DeleteCommunityCategory) (*Co
+
+	communityToValidate := communityToValidate{
+		id:         community.ID(),
+		clock:      community.config.CommunityDescription.Clock,
```
Not clear which role the clock is playing here. I do not see it being used in the run loop, also this is a previous community clock and not a new one

cammellos replied:
We pull them from the database in reverse clock order, and as soon as one validates, we stop validating any other with the same id, good point about the clock, I'll fix it :)

cammellos commented:
@jrainville @osmaczko @endulab @mprakhov Updated and addressed feedback.

I have added a test that should cover the whole flow https://github.com/status-im/status-go/blob/9a5e86049e1d12dc61d9422d8d35ad6c36a71a34/protocol/communities_messenger_signers_test.go#L110 

Let me know if I should cover more cases etc

endulab commented on protocol/communities/manager.go:
```diff
@@ -4771,6 +4956,70 @@ func (m *Manager) createCommunityTokenPermission(request *requests.CreateCommuni
+	if err != nil {
+		return nil, err
+	}
+	keyBytes := common.PubkeyToHexBytes(&pk.PublicKey)
```
Maybe I do not see something but does it make sense to marshal pubkey here and then unmarshal in UpdateControlNode?

mprakhov commented on protocol/communities/community.go:
```diff
@@ -2312,6 +2312,31 @@ func (o *Community) ValidateEvent(event *CommunityEvent, signer *ecdsa.PublicKey
 
 func (o *Community) MemberCanManageToken(member *ecdsa.PublicKey, token *community_token.CommunityToken) bool {
-	return o.IsMemberOwner(member) || (o.IsMemberTokenMaster(member) &&
+	return o.IsMemberOwner(member) || o.IsControlNode() || (o.IsMemberTokenMaster(member) &&
```
I don't think that we need to do a check of `o.IsControlNode()` as controlNode must be by default and Owner. It if is not right - then there is a bug somewhere

mprakhov commented on protocol/communities_messenger_helpers_test.go:
```diff
@@ -103,11 +118,21 @@ func (c *CollectiblesServiceMock) GetAssetContractData(chainID uint64, contractA
 }
 
 func (c *CollectiblesServiceMock) SetMockCollectibleContractData(chainID uint64, contractAddress string, collectible *communitytokens.CollectibleContractData) {
+	fmt.Println("SETTING", collectible)
```
do we need this fmt.Println? (here and below)

osmaczko replied:
:broom:

mprakhov commented on protocol/communities_messenger_signers_test.go:
```diff
@@ -128,102 +154,93 @@ func (s *MessengerCommunitiesSignersSuite) TestControlNodeUpdate() {
 	s.Require().NoError(err)
 
-	// alice accepts community update
+	// We check the control node is correctly set on john
```
to alice

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -1325,21 +1467,55 @@ func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, des
 	// Workaround for https://github.com/status-im/status-desktop/issues/12188
 	HydrateChannelsMembers(types.EncodeHex(id), description)
 
+	// We should queue only if the community has a token owner, and the owner has been verified
```
```suggestion
	// We should queue only if the community has a token owner, and the owner has not been verified
```

mprakhov commented on protocol/communities_messenger_signers_test.go:
```diff
@@ -128,102 +154,93 @@ func (s *MessengerCommunitiesSignersSuite) TestControlNodeUpdate() {
+	s.Require().NoError(err)
+	s.Require().NotNil(johnCommunity)
+	s.Require().True(common.IsPubKeyEqual(johnCommunity.ControlNode(), &s.alice.identity.PublicKey))
+	s.Require().False(johnCommunity.IsControlNode())
```
need to check, that Alice became an Owner, as the control node always has a role Owner
and john should loose his role and get it based on the community permission

cammellos replied:
that is a bit less clear, we discussed with @osmaczko , but it's alice responsibility to then update the description and remove John from the member list as having token ownership (as I understand, there's some cases where we might want to have a separate owner from control node), this PR is not currently addressing that though (maybe in a separate?)

mprakhov replied:
Agree, not in this. I have PR with kicking all members

osmaczko commented on protocol/communities_messenger_helpers_test.go:
```diff
@@ -95,6 +109,7 @@ func (c *CollectiblesServiceMock) GetCollectibleContractData(chainID uint64, con
 }
 
 func (c *CollectiblesServiceMock) GetAssetContractData(chainID uint64, contractAddress string) (*communitytokens.AssetContractData, error) {
+	fmt.Println("CHAINID", chainID, "contract", contractAddress, c.Assets)
```
:broom:

osmaczko commented on protocol/communities_messenger_helpers_test.go:
```diff
@@ -103,11 +118,21 @@ func (c *CollectiblesServiceMock) GetAssetContractData(chainID uint64, contractA
+}
+
+func (c *CollectiblesServiceMock) SafeGetSignerPubKey(ctx context.Context, chainID uint64, communityID string) (string, error) {
+	fmt.Println("GETTING SIGNER PUBKEY", c.Signers)
```
:broom:

osmaczko commented on protocol/communities_messenger_signers_test.go:
```diff
@@ -0,0 +1,246 @@
+
+	var published bool
+
+	// We check the updated description is published as a signal
```
Why do we check it as signal? Shouldn't John and Bob populate `MessengerResponse` with updated community in `RetrieveAll`?

cammellos replied:
no, maybe I misunderstand the question, but in this case, the check is asynchronous (it's queued), and so it won't be published through RetrieveAll, but as a signal.
Did I understand the question correctly?

osmaczko replied:
Ah, you are right :+1:

FYI: I have created a helper for that case: https://github.com/status-im/status-go/blob/19771c422da460e870aa53701589be80cde1ff52/protocol/messenger_testing_utils.go#L53 but the PR with that is in deep draft. I'll use it here once I merge it.

cammellos replied:
neat, we need to test more signals!

osmaczko commented on protocol/messenger_communities.go:
```diff
@@ -231,6 +233,25 @@ func (m *Messenger) handleCommunitiesHistoryArchivesSubscription(c chan *communi
 				if sub.ImportingHistoryArchiveMessagesSignal != nil {
 					m.config.messengerSignalsHandler.ImportingHistoryArchiveMessages(sub.ImportingHistoryArchiveMessagesSignal.CommunityID)
 				}
+				if sub.TokenCommunityValidated != nil {
```
Not sure if it belongs to `handleCommunitiesHistoryArchivesSubscription`

osmaczko commented on protocol/messenger_communities.go:
```diff
@@ -297,6 +318,7 @@ func (m *Messenger) handleCommunitiesSubscription(c chan *communities.Subscripti
 					return
 				}
 				if sub.Community != nil {
+					fmt.Println("PUBLISHING ORG", sub.Community.Description().TokenPermissions)
```
:broom:

mprakhov commented on protocol/communities/manager.go:
```diff
@@ -360,7 +377,82 @@ func (m *Manager) runENSVerificationLoop() {
+	}()
+}
+
+func (m *Manager) runOwnerVerificationLoop() {
```
could you please check `TestTokenMasterReceiveOwnerTokenFromControlNode` test?
I was unable to make it work after the control node deployed ownerToken and owner permission
It seems this loop is not starting
P.S. I did additional set up for test case:
```
s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.controlNode.identity.PublicKey))
communities.SetValidateInterval(50 * time.Millisecond)
```

cammellos replied:
I need to fix the test, yes, it's because you don't get it through retrieveall, but it will be published eventually as a signal, so I think I need to adjust the test, I will look into it, well spotted, thank you

cammellos commented:
@endulab @osmaczko @mprakhov @jrainville I have addressed (I hope I haven't missed any) the feedback, it should be ready to be reviewed again, thank you!

osmaczko commented:
Massive work, thanks @cammellos :rocket:

mprakhov commented:
Could you please help a little: how can I listen for a signal in tests (do we have a working example)? Maybe I will have a possibility to help you with fixing the tests

cammellos commented:
> Could you please help a little: how can I listen for a signal in tests (do we have a working example)? Maybe I will have a possibility to help you with fixing the tests

I think the test that was broken now is testing a bit less (we only validate that the token is received, but we don't check the exact type), but the test I have added covers that flow completely (a token owner is sent, and we make sure it's received, and processed correctly). Or there's still some concern on your part?

mprakhov commented:
> > Could you please help a little: how can I listen for a signal in tests (do we have a working example)? Maybe I will have a possibility to help you with fixing the tests
> 
> I think the test that was broken now is testing a bit less (we only validate that the token is received, but we don't check the exact type), but the test I have added covers that flow completely (a token owner is sent, and we make sure it's received, and processed correctly). Or there's still some concern on your part?

No concerns, if you think it is covered already, it can be outdated in this case and we can remove it
I just wonder if `TestTokenMasterReceiveOwnerTokenFromControlNode` pass?

osmaczko commented:
> Could you please help a little: how can I listen for a signal in tests (do we have a working example)? Maybe I will have a possibility to help you with fixing the tests

https://github.com/status-im/status-go/pull/4108#discussion_r1358317429

cammellos commented:
> No concerns, if you think it is covered already, it can be outdated in this case and we can remove it
I just wonder if TestTokenMasterReceiveOwnerTokenFromControlNode pass?

Oh I missed that one, I updated it in the same way, so we check that the community description as in queued, similarly to the other test that was failing, the full flow it should be covered in the new test (otherwise we would have to stub/mock all the interactions with blockchain)

mprakhov reviewed (approved):
Looks great 👍

jrainville reviewed (approved):
Awesome work! Let's ship it! :rocket:

mprakhov commented on protocol/communities/manager.go:
```diff
@@ -1355,12 +1475,42 @@ func (m *Manager) DeleteCategory(request *requests.DeleteCommunityCategory) (*Co
+	return nil
+}
+
+func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, description *protobuf.CommunityDescription, payload []byte, shard *common.Shard, verifiedOwner *ecdsa.PublicKey) (*CommunityResponse, error) {
```
please, double-check who is calling this fn...
I have a crash in one of my tests in `handleChatMessage`, `m.handleWrappedCommunityDescriptionMessage` returns nil due to a queue
